### PR TITLE
Calculate XCM fee

### PR DESCRIPTION
**Pull request by rhuttman**

### CODE REVIEW

File: pallets/xcmp-handler/src/lib.rs [hunk 1]
```diff
@@ -154,4 +164,24 @@ pub mod pallet {
...
+		pub fn calculate_xcm_fee(
+			para_id: ParachainId,
+			currency_id: T::CurrencyId,
+			transact_encoded_call_weight: u64,
+		) -> Result<u128, DispatchError> {
+			if let Some(xcm_data) = XcmChainCurrencyData::<T>::get(para_id, currency_id) {
+				if let Some(raw_fee) =
+					xcm_data.fee_per_second.checked_mul(transact_encoded_call_weight as u128)
+				{
+					Ok(raw_fee / (WEIGHT_PER_SECOND as u128))
```
Comment (justinzhou93) [inline pallets/xcmp-handler/src/lib.rs:178]: saturating/checked div prob not necessary, but is it good practice to keep consistent?

**Review by justinzhou93 — APPROVED** [review]
1 nit

File: pallets/xcmp-handler/src/lib.rs (hunk 1 above)
Comment (rhuttman) [reply pallets/xcmp-handler/src/lib.rs:178]: I can't think of why I would need that so I left it out

File: pallets/xcmp-handler/src/lib.rs [hunk 2]
```diff
@@ -154,4 +164,24 @@ pub mod pallet {
...
+				} else {
+					Err(Error::<T>::FeeOverflow)?
+				}
+			} else {
+				Err(Error::<T>::CurrencyChainComboNotFound)?
+			}
```
Comment (nvengal) [inline pallets/xcmp-handler/src/lib.rs:184]: I've been trying to write more "idiomatic" rust, but not sure that I actually get it yet.
```suggestion
			XcmChainCurrencyData::<T>::get(para_id, currency_id).map_or(
				Err(Error::<T>::CurrencyChainComboNotFound.into()),
				|xcm_data| {
					xcm_data
						.fee_per_second
						.checked_mul(transact_encoded_call_weight as u128)
						.ok_or(Error::<T>::FeeOverflow.into())
						.map(|raw_fee| raw_fee / (WEIGHT_PER_SECOND as u128))
				},
			)
```

**Review by nvengal — APPROVED** [review]

File: pallets/xcmp-handler/src/tests.rs [hunk 3]
```diff
@@ -128,3 +128,64 @@ fn errors_if_not_found() {
...
+		let xcm_data = XcmCurrencyData {
+			native: false,
+			fee_per_second: 50_000_000_000,
+			instruction_weight: 1_000,
+		};
+		XcmChainCurrencyData::<Test>::insert(para_id, currency_id, xcm_data.clone());
+		let transact_encoded_call_weight: u64 = 100_000_000;
+
+		let expected_fee = xcm_data.fee_per_second * (transact_encoded_call_weight as u128) /
+			(WEIGHT_PER_SECOND as u128);
```
Comment (atomaka) [inline pallets/xcmp-handler/src/tests.rs:151]: It'd be nice if this were just the static value for simplicity, but I guess `WEIGHT_PERSECOND` could potentially change and is non-trivial to stub.

**Review by atomaka — APPROVED** [review]

**Pull request merged**